cardano-db/src/Cardano/Db/Operations/Core/MinId.hs

```diff
@@ -0,0 +1,165 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module Cardano.Db.Operations.Core.MinId where
+
+import Cardano.Db.Operations.Core.Query (queryMinRefId)
+import Cardano.Db.Operations.Types (MaTxOutFields (..), TxOutFields (..), TxOutTableType (..))
+import Cardano.Db.Schema.BaseSchema
+import qualified Cardano.Db.Schema.Core.TxOut as C
+import qualified Cardano.Db.Schema.Variant.TxOut as V
+import Cardano.Prelude
+import qualified Data.Text as Text
+import Database.Persist.Sql (PersistEntity, PersistField, SqlBackend, fromSqlKey, toSqlKey)
+
+data MinIds (a :: TxOutTableType) = MinIds
+  { minTxInId :: Maybe TxInId
+  , minTxOutId :: Maybe (TxOutIdFor a)
+  , minMaTxOutId :: Maybe (MaTxOutIdFor a)
+  }
```

Comment:
There is a lot of code duplication in this module, could it be reduced by using the `TxOutFields` class more? Alternatively MinIds could be defined using `TxOutIdW` and `TxOutIdW` or even by introducing a `MinIdsW` wraper.

Reply:
I was skeptical as to what I should do as I don't really understand the original implementation I was just trying to keep original functionality and have `minTxOutId` + `minMaTxOutId` be variants